django_ratelimit/decorators.py

```diff
@@ -5,29 +5,44 @@
 
 from django_ratelimit import ALL, UNSAFE
 from django_ratelimit.exceptions import Ratelimited
-from django_ratelimit.core import is_ratelimited
+from django_ratelimit.core import is_ratelimited, ais_ratelimited
+from asgiref.sync import iscoroutinefunction
 
 
 __all__ = ['ratelimit']
 
 
 def ratelimit(group=None, key=None, rate=None, method=ALL, block=True):
     def decorator(fn):
-        @wraps(fn)
-        def _wrapped(request, *args, **kw):
-            old_limited = getattr(request, 'limited', False)
-            ratelimited = is_ratelimited(request=request, group=group, fn=fn,
-                                         key=key, rate=rate, method=method,
-                                         increment=True)
-            request.limited = ratelimited or old_limited
-            if ratelimited and block:
-                cls = getattr(
-                    settings, 'RATELIMIT_EXCEPTION_CLASS', Ratelimited)
-                raise (import_string(cls) if isinstance(cls, str) else cls)()
-            return fn(request, *args, **kw)
+        if iscoroutinefunction(fn):
```

Comment:
Not something to change here, but I wonder if having a second `aratelimit` decorator might be simpler / clearer. I know we've only got 3 months of 3.2 official support left, but I think it might make it easier to say "`aratelimit` requires Django >= 4". "Explicit is better than implicit," after all

Reply:
Async is barely complete even in Django 5.0, so doubt anybody would care if 3.2 doesn't support the async decorator. If you're doing async work, you're probably running the latest version of Django.